### PR TITLE
SE-748-Change-Atomic-Rule-Set-functions-in-token-handlers-to-accept-ActionTypes-Array-length-of-0

### PR DESCRIPTION
**Pull request by Palmerg4**
Removed check for 0 length actions array input argument from all atomic full rule set functions. Added tests for each.

### CODE REVIEW

File: src/client/token/handler/ruleContracts/HandlerAccountApproveDenyOracle.sol [hunk 1]
```diff
@@ -37,7 +37,6 @@ contract HandlerAccountApproveDenyOracle is RuleAdministratorOnly, ActionTypesAr
      * @param _ruleIds Rule Id corresponding to the actions
      */
     function setAccountApproveDenyOracleIdFull(ActionTypes[] calldata _actions, uint32[] calldata _ruleIds) external ruleAdministratorOnly(lib.handlerBaseStorage().appManager) {
-        if (_actions.length == 0) revert InputArraysSizesNotValid();
         if (_actions.length != _ruleIds.length) revert InputArraysMustHaveSameLength();
         clearAccountApproveDenyOracle();
```
Comment (VoR0220) [inline src/client/token/handler/ruleContracts/HandlerAccountApproveDenyOracle.sol:42]: Is this the portion doing the clearing in lieu of the cancelling of the zero length arrays?

File: src/client/token/handler/ruleContracts/HandlerAccountMaxTradeSize.sol [hunk 2]
```diff
@@ -43,7 +43,6 @@ contract HandlerAccountMaxTradeSize is RuleAdministratorOnly, ActionTypesArray,
      * @param _ruleIds Rule Id corresponding to the actions
      */
     function setAccountMaxTradeSizeIdFull(ActionTypes[] calldata _actions, uint32[] calldata _ruleIds) external ruleAdministratorOnly(lib.handlerBaseStorage().appManager) {
-        if (_actions.length == 0) revert InputArraysSizesNotValid();
         if (_actions.length != _ruleIds.length) revert InputArraysMustHaveSameLength();
         clearAccountMaxTradeSize();
```
Comment (VoR0220) [inline src/client/token/handler/ruleContracts/HandlerAccountMaxTradeSize.sol:48]: Same question here

File: src/client/token/handler/ruleContracts/HandlerAccountApproveDenyOracle.sol (hunk 1 above)
Comment (Palmerg4) [reply src/client/token/handler/ruleContracts/HandlerAccountApproveDenyOracle.sol:42]: These clearing internals were always present. DOOM just needs a way to call this internal by passing zero length arrays so they can clear rules as needed without replacing them

**Review by TJ-Everett — APPROVED** [review]
Looks good to me. Just a suggestion to add comments in the natspec block for why we do not check for 0 length in these for future audits

**Review by VoR0220 — APPROVED** [review]
GTM

**Review by ShaneDuncan602 — APPROVED** [review]
Looks fantastic.

**Pull request merged**